stdlib/string.gr

```diff
@@ -1609,3 +1609,64 @@ export let rec forEachCodePointi = (fn: (Number, Number) -> Void, str: String) =
   Memory.decRef(WasmI32.fromGrain(forEachCodePointi))
   void
 }
+
+/**
+ * Trims a string removing any leading or following white spaces based on the end param.
+ *
+ * @param str: The string to be trimmed.
+ * @param end: If true we are trimming the following spaces, otherwise trim leading spaces.
+ * @returns The trimmed string.
+ * 
+ * @example String.trimString("   Hello World   ", false)
+ */
+let trimString = (str: String, end: Bool) => {
```

Comment:
@grain-lang/core Thoughts on `trim` vs `strip`?

Reply:
I think Python and Ruby call it `strip` and just about everyone else calls it `trim`. I personally prefer `trim`, since when I see `strip` I think of it as being possible throughout the string. Sorta like `deburr`, honestly. Or that I get to choose character classes of what I want to strip. Whereas I like `trim` since it feels more bounded in what it could possibly do.

Maybe doing a ton of JS dev has just molded me, idk.

Reply:
I don't know how much of a difference my input makes here but personally, I agree with @ospencer,  strip sounds a lot more like you are removing a value from the string no matter the position whereas trim sounds is more like you are trimming off extra stuff be it the start or the beginning.

Reply:
I agree with `trim`